client/post-editor/editor-sidebar/header.jsx

```diff
@@ -22,19 +24,32 @@ import DraftsButton from 'post-editor/drafts-button';
 import PostCountsData from 'components/data/post-counts-data';
 import QueryPostTypes from 'components/data/query-post-types';
 
+function goBack() {
+	if ( typeof window !== 'undefined' ) {
+		window.history.back();
+	}
+}
+
 function EditorSidebarHeader( { typeSlug, type, siteId, showDrafts, toggleDrafts, allPostsUrl, toggleSidebar } ) {
 	const isCustomPostType = ( 'post' !== typeSlug && 'page' !== typeSlug );
 	const className = classnames( 'editor-sidebar__header', {
 		'is-drafts-visible': showDrafts,
 		'is-loading': isCustomPostType && ! type
 	} );
-
-	let allPostsLabel;
-	switch ( typeSlug ) {
-		case 'post': allPostsLabel = translate( 'Posts' ); break;
-		case 'page': allPostsLabel = translate( 'Pages' ); break;
-		default: allPostsLabel = get( type, 'labels.menu_name' ) || translate( 'Loading…' );
+	const useBackButton = page.len > 0;
```

Comment:
If I follow correctly, this changes the behavior such that if the user arrives at the editor from any page, the header displays a "Back" button instead of "Posts"? I'm on the fence with this one, but I'm not strictly opposed to it outside of the concern that it breaks existing expectations. Care to weigh in, @mtias or @folletto ?